website/public/js/services/userServices.js

```diff
@@ -259,12 +266,19 @@ angular.module('habitrpg')
         },
 
         unlock: function (data) {
-          $window.habitrpgShared.ops['unlock'](user, data);
           callOpsFunctionAndRequest('unlock', 'unlock', "POST", '', data);
         },
 
         set: function(updates) {
           setUser(updates);
+          $http({
+            method: "PUT",
+            url: 'api/v3/user',
+            data: updates,
+          })
+          .then(function (response) {
+            sync();
```

Comment:
So, it looks like the for user loop doesn't work. You can't set a object's sub object properties with string notation without a work around. 

I can add this workaround: http://stackoverflow.com/questions/6491463/accessing-nested-javascript-objects-with-string-key .

Or, I can update all the User.set calls to actually set the user object before calling User.set.

For example:
```
$scope.addGold = function(){
      User.user.stats.gp = User.user.stats.gp + 500;
      User.set({
        'stats.gp': User.user.stats.gp + 500,
      });
    };
```